pipeline.py

```diff
@@ -51,19 +58,25 @@ class CorpusManager:
     """
 
     def __init__(self, path_to_raw_txt_data: str):
-        pass
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self._storage = {}
+        self._scan_dataset()
 
     def _scan_dataset(self):
         """
         Register each dataset entry
         """
-        pass
+        path = Path(self.path_to_raw_txt_data)
+
+        for file in path.rglob('*_raw.txt'):
+            article_id = int(file.parts[-1].split('_')[0])
```

Comment:
1. there is a better way to get a file name (instead of using last of its parts)
2. this is too dangerous to think that there will be always an integer

Reply:
done